bridge_common/validate_job_api.py

```diff
@@ -0,0 +1,243 @@
+        if 'sds_type' not in apiJob:
+            return (False, "Invalid request. Api Job sds type not found")
+        if 'object_type' not in apiJob:
+            return (False, "Invalid request. Api Job object type not found")
+        return (True, "")
+
+    def _checkCustomType(self, customType, inputParm, inputVal):
+        def _check(inputVal, cType=customType):
+            # check whether the custom defined type is decleared!
+            typeDef = self.yamlObj.get(cType.lower())
+            if not typeDef:
+                return (False, "yaml custom type: '%s' details not found" % (
+                    cType))
+
+            # check whether the item(s) type is valid
+            for item in inputVal:
+                itemType = typeDef['attrs'].get(inputParm, {}).get('type')
+                if not PRIMITIVE_TYPES[itemType](item):
+                    return (False, "Invalid parameter type: "
+                            + "%s. Expected value type is: %s" % (
+                                item, itemType))
+            return (True, "")
+
+        # customType is an array
+        if customType.endswith('[]'):
+            # check whether the given input is an array
+            if not PRIMITIVE_TYPES['List'](inputVal):
+                return (False, "Invalid input type: %s. " % (
+                    inputVal) + " Expected value type is an array")
+            return _check(inputVal, customType[:-2])
+        else:
+            # Its a non-array custom type
+            return _check([inputVal])
+
+    def _getTag(self, dotPath):
+        path = self.yamlObj
+        for p in dotPath.split('.'):
+            path = path.get(p, {})
+        return path
+
+    def validateApi(self, apiJob):
+        if not self.yamlObj:
+            return (False, "Validating schema not loaded!")
+
+        status = self._validateApiJob(apiJob)
+        if not status[0]:
+            return status
+
+        # Check object type defined
+        if apiJob['object_type'] not in self.yamlObj:
+            return (False, "Given object type:%s not defined in the yaml" % (
+                apiJob['object_type']))
+
+        # Check flow exists
+        flow = self.yamlObj['flows'].get(apiJob.get('flow'))
+        if not flow:
+            return (False, "Flow: %s not defined in the yaml flows" % (
+                apiJob['flow']))
+
+        # A flow can have multiple automs defined
+        # Validate list of atoms assigned for the flow
+        for fatms in flow['atoms']:
+            # check whether specified atom in the flows are found in atoms
+            atom = self._getTag(fatms)
+            if not atom:
+                return (False, "Atom: %s not found for the flow: %s" % (
+                    fatms, apiJob.get('flow')))
+
+            # check whether given flow is defined in the atoms
+            # atom = self.yamlObj[apiJob['object_type']]['atoms']
+            if apiJob['flow'] not in atom.get('flows'):
```

Comment:
There is a change in how we get the "flow" 
Refer: https://github.com/Tendrl/bridge_common/blob/master/etc/samples/tendrl_definitions_generic-0.1.sample.yaml

1) Take the "attributes".keys() unique list from the tendrl api job
2) Take the list of "flows" supported by an atom (https://github.com/Tendrl/bridge_common/blob/master/etc/samples/tendrl_definitions_generic-0.1.sample.yaml#L51)
3) For each of these supported flows, match the "attributes" list from step 1 with the "inputs" list of the flow (https://github.com/Tendrl/bridge_common/blob/master/etc/samples/tendrl_definitions_generic-0.1.sample.yaml#L81)
4. Result from step 3 is your "flow" for this api job